models.py

```diff
@@ -105,162 +122,94 @@ class CustOrderStatus(Base):
-                        filter(DeliveryPerson.delivery_person_id == CustOrderStatus.delivery_person_id).\
-                        filter(CustOrderStatus.order_id == order_id)
-            return view
-        except Exception as ex:
-            print("Error getting order, error={}".format(str(ex)))
-        finally:
-            session.close()
-
-
-    def view_sales_today(self, order_status, session):
-        """ View sales of today """
-
-        try:
-            sales_today = session.query(CustOrderStatus, CustomerDetails).\
-                            filter(CustomerDetails.cust_id == CustOrderStatus.cust_id).\
-                            filter(CustOrderStatus.order_status == order_status).\
-                            filter(CustOrderStatus.checkout_time <= datetime.today()).\
-                            group_by(CustOrderStatus.order_id)
-            return sales_today
-        except Exception as ex:
-            print("Error getting order, error={}".format(str(ex)))
-        finally:
-            session.close()
-
-
-    def sum_revenue_today(self, order_status, session):
-        """ Sum revenue of today """
-
-        try:
-            revenue = session.query(CustOrderStatus, func.sum(CustOrderStatus.bill_amount)).\
-                        filter(CustOrderStatus.order_status == order_status).\
-                        filter(CustOrderStatus.checkout_time <= datetime.today())
-            return revenue
-        except Exception as ex:
-            print("Error getting order, error={}".format(str(ex)))
-        finally:
-            session.close()
-
-
-    def delete_order(self, order_id, session):
+        view = session.query(CustomerDetails, CustOrderStatus, DeliveryPerson).\
+                filter(CustomerDetails.cust_id == CustOrderStatus.cust_id).\
+                filter(DeliveryPerson.delivery_person_id == CustOrderStatus.delivery_person_id).\
+                filter(CustOrderStatus.order_id == order_id)
+        return view
+        
+    def view_revenue_today(self, session, order_status):
+        """ View revenue/sales of today """
+
+        sales_today = text("""SELECT cd.cust_name, cos.order_id, cos.order_status, cos.bill_amount, cos.checkout_time 
+            FROM customer_order_status as cos 
+            JOIN customer_details as cd 
+                ON cd.cust_id = cos.cust_id 
+            WHERE cos.order_status = {} AND date(cos.checkout_time) = date(CURRENT_DATE);""".format(order_status))
+        result = session.connection().execute(sales_today).fetchall()
+        return result
+            
+    def sum_revenue_today(self, session, order_status):
+        """ Sum revenue/sales of today """
+
+        rev = text("""SELECT sum(cos.bill_amount) 
+            FROM customer_order_status as cos 
+            WHERE cos.order_status = {} AND date(cos.checkout_time) = date(CURRENT_DATE);""".format(order_status))
+        result = session.connection().execute(rev).fetchall()
+        return result
```

Comment:
What's the difference b/w `view_revenue_today(...)` and `sum_revenue_today(...)`?

Reply:
changed view_revenue_today() to view_sales_today()